src/components/BooksQuery/BooksQuery.tsx

```diff
@@ -1,13 +1,35 @@
-import { SyntheticEvent, useState } from 'react'
+/* eslint-disable prefer-const */
+import { SyntheticEvent, useEffect, useState } from 'react'
 import './BooksQuery.css'
 import { Link } from 'react-router-dom'
-import { useGetBooks } from '../../hooks/useGetBooks'
+import { useAppSelector, useAppDispatch } from '../../redux/hook'
+import { GoogleBooks, selectCachedData } from '../../redux/BookSlice'
+import { BooksApiResponse } from '../../types/GoogleBooks'
 const BooksQuery = () => {
   const [bookName, setBookName] = useState<string>('')
```

Comment:
The inital state for "bookName" should be "searchInput" from the store, not an empty string